jina/clients/base/http.py

```diff
@@ -93,80 +93,59 @@ async def _get_results(
+                :return: asyncio Task for sending message
+                """
+                return asyncio.ensure_future(iolet.send_message(request=request))
+
+            def _result_handler(result):
+                return result
+
+            streamer = RequestStreamer(
+                self.args,
+                request_handler=_request_handler,
+                result_handler=_result_handler,
+            )
+            async for response in streamer.stream(request_iterator):
+                r_status = response.status
 
-                proto = 'https' if self.args.tls else 'http'
-                url = f'{proto}://{self.args.host}:{self.args.port}/post'
-                iolet = await stack.enter_async_context(
-                    HTTPClientlet(url=url, logger=self.logger)
-                )
+                r_str = await response.json()
+                self._handle_response_status(r_status, r_str, url)
 
-                def _request_handler(request: 'Request') -> 'asyncio.Future':
-                    """
-                    For HTTP Client, for each request in the iterator, we `send_message` using
-                    http POST request and add it to the list of tasks which is awaited and yielded.
-                    :param request: current request in the iterator
-                    :return: asyncio Task for sending message
-                    """
-                    return asyncio.ensure_future(iolet.send_message(request=request))
-
-                def _result_handler(result):
-                    return result
-
-                streamer = RequestStreamer(
-                    self.args,
-                    request_handler=_request_handler,
-                    result_handler=_result_handler,
-                )
-                async for response in streamer.stream(request_iterator):
-                    r_status = response.status
-
-                    r_str = await response.json()
-                    self._handle_response_status(r_status, r_str, url)
-
-                    da = None
-                    if 'data' in r_str and r_str['data'] is not None:
-                        from docarray import DocumentArray
-
-                        da = DocumentArray.from_dict(r_str['data'])
-                        del r_str['data']
-
-                    resp = DataRequest(r_str)
-                    if da is not None:
-                        resp.data.docs = da
-
-                    callback_exec(
-                        response=resp,
-                        on_error=on_error,
-                        on_done=on_done,
-                        on_always=on_always,
-                        continue_on_error=self.continue_on_error,
-                        logger=self.logger,
-                    )
-                    if self.show_progress:
-                        p_bar.update()
-                    yield resp
-
-            except (aiohttp.ClientError, ValueError, ConnectionError) as e:
```

Comment:
Please do not change this, the `Exception` should be properly handled. @JohannesMessner

Reply:
no, they should bn raised